### PR TITLE
Azexp waol

### PR DESCRIPTION
**Pull request by Eskibear**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/serviceexplorer/azure/webapps/OpenWebAppAction.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import com.microsoft.tooling.msservices.serviceexplorer.azure.webapp.LinuxWebAppNode;
+import com.microsoft.tooling.msservices.serviceexplorer.azure.webapp.WebAppNode;
+import com.microsoft.tooling.msservices.serviceexplorer.azure.webapp.WinWebAppNode;
+
+import java.awt.Desktop;
+import java.io.IOException;
+import java.net.URI;
+
+@Name("Open in Browser")
+public class OpenWebAppAction extends NodeActionListener {
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/serviceexplorer/azure/webapps/OpenWebAppAction.java:18]: extend from azure wrapper listener?

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/LinuxWebAppNode.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    @Override
+    public String getRunState() {
+        return webApp == null ? siteInner.state() : webApp.state();
+    }
+
+    @Override
+    public WebApp getWebApp() {
+        if (webApp == null) {
+            try {
+                return AzureWebAppMvpModel.getInstance().getWebAppById(subscriptionId, siteInner.id());
```
Comment (adashen) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/LinuxWebAppNode.java:62]: assign to webApp?

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/LinuxWebAppNode.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+        return webApp;
+    }
+
+    @Override
+    public Map<String, String> toProperties() {
+        final Map<String, String> properties = new HashMap<>();
```
Comment (adashen) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/LinuxWebAppNode.java:72]: more telemetry properties?

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/LinuxWebAppNode.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+import com.microsoft.azuretools.core.mvp.model.ResourceEx;
+import com.microsoft.azuretools.core.mvp.model.webapp.AzureWebAppMvpModel;
+import com.microsoft.azuretools.telemetry.AppInsightsConstants;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class LinuxWebAppNode extends WebAppNode {
+    private final SiteInner siteInner;
```
Comment (adashen) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/LinuxWebAppNode.java:36]: add presenter for this class?

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/WebAppModule.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+    }
+
+    /**
+     * Create nodes for both Windows and Linux Web Apps.
+     *
+     * @param winapps   list of Windows Web Apps
+     * @param linuxapps list of Linux Web Apps
+     */
+    public void renderWebApps(List<ResourceEx<WebApp>> winapps, List<ResourceEx<SiteInner>> linuxapps) {
```
Comment (adashen) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/WebAppModule.java:48]: move to presenter

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/WebAppNode.java [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+            super(WebAppNode.this,
+                    String.format("This operation will delete Web App %s.\nAre you sure you want to continue?",
+                            getWebAppName()),
+                    "Deleting Web App");
+        }
+
+        @Override
+        protected void azureNodeAction(NodeActionEvent e) throws AzureCmdException {
+            try {
+                AzureWebAppMvpModel.getInstance().deleteWebApp(subscriptionId, getWebAppId());
```
Comment (adashen) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/WebAppNode.java:117]: move this to presenter

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/serviceexplorer/azure/webapps/OpenWebAppAction.java (hunk 1 above)
Comment (Eskibear) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/serviceexplorer/azure/webapps/OpenWebAppAction.java:18]: no need, already has telemetry

**Pull request closed without merging**